legate/tester/stages/test_stage.py

```diff
@@ -230,7 +245,16 @@ def run(
         stage_args = self.args + self.shard_args(shard, config)
         file_args = self.file_args(test_file, config)
 
-        cmd = [str(config.legate_path), str(test_path)]
+        if config.cov_bin:
+            cmd = (
+                [str(config.legate_path)]
+                + [config.cov_bin]
+                + self.cov_args(config)
+                + [str(test_path)]
+            )
+        else:
+            cmd = [str(config.legate_path), str(test_path)]
+
         cmd += stage_args + file_args + config.extra_args
```

Comment:
If you change `cov_args` to include `config.cov_bin` (and return [] if `config.cov_bin` is None) then I think all this can be streamlined:

```python
cov_args = self.cov_args(config)  

cmd = str(config.legate_path)] + cov_args + [str(test_path)] 

cmd += stage_args + file_args + config.extra_args
``` 

I would like to preserve that the final assembly of `cmd` be an unconditional sum, so that that overall structure and order of the arguments stays immediately visible

Reply:
Good suggestion. Fixed. Thanks.